src/main/scala/dotty/dokka/tasty/TypesSupport.scala

```diff
@@ -0,0 +1,125 @@
+            //         convertTypeOrBoundsToReference(reflect)(qual) match {
+            //             case TypeReference(label, link, xs, _) => TypeReference(sym.name, link + "/" + label, xs, true)
+            //             case EmptyReference if sym.name == "<root>" | sym.name == "_root_" => EmptyReference
+            //             case EmptyReference => TypeReference(sym.name, "", Nil, true)
+            //             case _ => throw Exception("Match error in SymRef/TypeOrBounds/ClassDef. This should not happen, please open an issue. " + convertTypeOrBoundsToReference(reflect)(qual))
+            //         }
+
+            //         // NOTE: This branch handles packages, which are now TypeRefs
+            //         case sym if sym.isTerm || sym.isTypeDef =>
+            //         convertTypeOrBoundsToReference(reflect)(qual) match {
+            //             case TypeReference(label, link, xs, _) => TypeReference(sym.name, link + "/" + label, xs)
+            //             case EmptyReference if sym.name == "<root>" | sym.name == "_root_" => EmptyReference
+            //             case EmptyReference => TypeReference(sym.name, "", Nil)
+            //             case _ => throw Exception("Match error in SymRef/TypeOrBounds/Other. This should not happen, please open an issue. " + convertTypeOrBoundsToReference(reflect)(qual))
+            //         }
+            //         case sym => throw Exception("Match error in SymRef. This should not happen, please open an issue. " + sym)
+            //     }
+            //     case _ =>
+            //     throw Exception("Match error in TypeRef. This should not happen, please open an issue. " + convertTypeOrBoundsToReference(reflect)(qual))
+            // }
+        case TermRef(qual, typeName) =>
+            // convertTypeOrBoundsToReference(reflect)(qual) match {
+            //     case TypeReference(label, link, xs, _) => TypeReference(typeName + "$", link + "/" + label, xs)
+            //     case EmptyReference => TypeReference(typeName, "", Nil)
+            //     case _ => throw Exception("Match error in TermRef. This should not happen, please open an issue. " + convertTypeOrBoundsToReference(reflect)(qual))
+            // }
+            noSupported("TypeRef") 
+
+        // NOTE: old SymRefs are now either TypeRefs or TermRefs - the logic here needs to be moved into above branches
+        // NOTE: _.symbol on *Ref returns its symbol
+        // case SymRef(symbol, typeOrBounds) => symbol match {
+        // }
+        // case _ => throw Exception("No match for type in conversion to Reference. This should not happen, please open an issue. " + tp)
+        case TypeBounds(low, hi) =>
+            typeBound(low, low = true) ++ typeBound(low, low = false)
+        
+        case reflect.NoPrefix() => Nil
+
+    private def typeBound(t: Type, low: Boolean) = 
+        val ignore = if(low) t.typeSymbol == defn.NothingClass  else t.typeSymbol == defn.AnyClass
```

Comment:
We should have a test that goes into the bounds of an HKT (i.e. a parameter like `F[+X] <: List[X]`). They need special rendering code, but also they may have different "empty" bounds from `Nothing`, `Any` (IIRC at this point empty bounds are always Nothing and Any, but we were planning to change that).